### PR TITLE
fix: Refactor Event SQL to use named parameters - Events grid - [ DHIS2-13508 ]

### PR DESCRIPTION
**Pull request by lucaCambi77**
https://jira.dhis2.org/browse/DHIS2-13508
The final step for named parameters:

- Add named parameter to Events grid. Overall clean code to make it more even with event SQL.
- Add AssertAll and parameterized tests which run for common cases of event and event gird
- Temporal binding (dates and timestamp)

For some reason, the first commit messed up the formatting although checks passed. Proper formatting is restored later

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java [hunk 1]
```diff
@@ -173,23 +171,18 @@ public class JdbcEventStore implements EventStore
...
-        " userinfo.uid                   as usernote_uid," +
-        " userinfo.username              as usernote_username," +
-        " userinfo.firstname             as userinfo_firstname," +
-        " userinfo.surname               as userinfo_surname" +
-        " from programstageinstancecomments psic" +
-        " inner join trackedentitycomment psinote" +
-        " on psic.trackedentitycommentid = psinote.trackedentitycommentid" +
-        " left join userinfo on psinote.lastupdatedby = userinfo.userinfoid ";
+    private static final String PSI_EVENT_COMMENT_QUERY = "select psic.programstageinstanceid    as psic_id,"
+        + " psinote.trackedentitycommentid as psinote_id," + " psinote.commenttext            as psinote_value,"
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java:175]: Looks like the format got messed up here. Is it possible to fix this?

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java [hunk 2]
```diff
@@ -1076,559 +1042,611 @@ private String getEventSelectQuery( EventSearchParams params, MapSqlParameterSou
...
+                        .append( " psi.lastupdated < " )
+                        .append( ":lastUpdatedEnd" )
+                        .append( "' " );
                 }
                 else
                 {
-                    sqlBuilder.append( hlp.whereAnd() ).append( " psi.lastupdated <= '" )
-                        .append( DateUtils.getLongDateString( params.getLastUpdatedEndDate() ) ).append( "' " );
+                    mapSqlParameterSource.addValue( "lastUpdatedEnd",
+                        DateUtils.getLongDateString( params.getLastUpdatedEndDate() ), Types.TIMESTAMP );
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java:1644]: Do we need getLongDateString here? I see you removed it from the other places and used Types.TIMESTAMP

**Review by stian-sandvold — CHANGES_REQUESTED** [review]

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java (hunk 1 above)
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java:175]: @stian-sandvold I literally have no idea what happened with the formatting. I've restored most of the previous formatting, some new changes were actually a bit better so I've left them

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java (hunk 2 above)
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java:1644]: ok thanks for spotting that was going to fail with merged tests

**Review by enricocolasante — APPROVED** [review]

**Review by stian-sandvold — APPROVED** [review]

**Pull request merged**